tests/TestCase/Core/ConfigureTest.php

```diff
@@ -592,4 +592,29 @@ public function testConsumeEmpty()
         $result = Configure::consume(null);
         $this->assertNull($result);
     }
+
+    /**
+     * testConsumeOrFail method
+     *
+     * @return void
+     */
+    public function testConsumeOrFail()
+    {
+        $expected = 'ok';
+        Configure::write('This.Key.Exists', $expected);
+        $result = Configure::consumeOrFail('This.Key.Exists');
+        $this->assertEquals($expected, $result);
+    }
```

Comment:
An assertion testing that the value has been consumed and cannot be read again would be nice.

Reply:
I think that part is covered in consume tests itself already.